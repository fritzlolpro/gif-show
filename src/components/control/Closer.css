.closer {
  position: relative;
  top: 40px;
  display: inline-block;
  background-color: #fff;
  border-radius: 50%;
  height: 68px;
  width: 68px;
}
.closer i {
  font-weight: 900;
  font-size: 32px;
}
.closer .icon {
  position: absolute;
  width: 32px;
  height: 32px;
  position: relative;
  top: 18px;
  left: 14px;
}
.closer .icon:before,
.closer .icon:after {
  position: absolute;
  content: ' ';
  height: 30px;
  width: 8px;
  background-color: #4a90e2;
}
.closer .icon:before {
  -webkit-transform: rotate(45deg);
          transform: rotate(45deg);
}
.closer .icon:after {
  -webkit-transform: rotate(-45deg);
          transform: rotate(-45deg);
}
.closer .icon:hover:before,
.closer .icon:hover:after {
  background-color: #1d77dd;
}
