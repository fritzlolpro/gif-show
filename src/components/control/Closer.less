.closer {
    position: relative;
    top: 40px;
    display: inline-block;
    background-color: #fff;
    border-radius: 50%;
    height: 68px;
    width: 68px;
    i {
        font-weight: 900;
        font-size: 32px;
    }
    .icon {
        position: absolute;
        width: 32px;
        height: 32px;
        position: relative;
        top: 18px;
        left: 14px;
        &:before,
        &:after {
            position: absolute;
            content: ' ';
            height: 30px;
            width: 8px;
            background-color: #4a90e2;
        }
        &:before {
            transform: rotate(45deg);
        }
        &:after {
            transform: rotate(-45deg);
        }
        &:hover {
            &:before,
            &:after {
                background-color: #1d77dd;
            }
        }
    }
}