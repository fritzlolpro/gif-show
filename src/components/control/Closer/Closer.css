.closer {
  position: relative;
  display: inline-block;
  background-color: #fff;
  border-radius: 50%;
  height: 68px;
  width: 68px;
}
@media only screen and (max-width : 479px) and (min-width : 321px), only screen and (max-width : 1023px) and (min-width : 480px) {
  .closer {
    top: 10px;
    height: 48px;
    width: 48px;
  }
}
.closer .icon {
  position: absolute;
  width: 32px;
  height: 32px;
  position: relative;
  top: 18px;
  left: 14px;
}
.closer .icon:before,
.closer .icon:after {
  position: absolute;
  content: ' ';
  height: 30px;
  width: 8px;
  background-color: #4a90e2;
}
@media only screen and (max-width : 479px) and (min-width : 321px), only screen and (max-width : 1023px) and (min-width : 480px) {
  .closer .icon:before,
  .closer .icon:after {
    height: 20px;
    width: 6px;
  }
}
.closer .icon:before {
  -webkit-transform: rotate(45deg);
          transform: rotate(45deg);
}
.closer .icon:after {
  -webkit-transform: rotate(-45deg);
          transform: rotate(-45deg);
}
.closer .icon:hover:before,
.closer .icon:hover:after {
  background-color: #1d77dd;
}
@media only screen and (max-width : 479px) and (min-width : 321px), only screen and (max-width : 1023px) and (min-width : 480px) {
  .closer .icon {
    top: 14px;
    left: 5px;
  }
}
