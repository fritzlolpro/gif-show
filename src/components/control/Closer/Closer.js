import React from 'react'
import {Link} from 'react-router-dom'
import './Closer.css'

const Closer = () => {
    return (
        <Link to="/">
        <div className='closer'>

            <div className="icon"></div>
        </div>
        </Link>
    )
}

export default Closer