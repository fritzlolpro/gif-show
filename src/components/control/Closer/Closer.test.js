import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter: new Adapter()})

import Closer from './Closer'

describe('Closer', () => {

    it('Defined', () => {
        expect(Closer).toBeDefined();
    });

    it('Renders', () => {
        const tree = shallow(<Closer/>);
        expect(tree).toMatchSnapshot();
    });
})

