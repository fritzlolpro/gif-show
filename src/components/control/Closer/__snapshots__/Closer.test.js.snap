// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Closer Renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Closer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="closer"
>
        <div
                className="icon"
        />
</div>,
      "replace": false,
      "to": "/",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="icon"
/>,
        "className": "closer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "icon",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="closer"
>
          <div
                    className="icon"
          />
</div>,
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="icon"
/>,
          "className": "closer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "icon",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
